### PR TITLE
UI plugin: vSphere MC credentials osImage check

### PR DESCRIPTION
**Pull request by swalner-vmware**

### CODE REVIEW

File: cli/cmd/plugin/ui/web/tanzu-ui/node-server/src/routes/api/endpoints/vsphere.js [hunk 1]
```diff
@@ -158,16 +158,24 @@ router.get('/resourcepools', (req, res) => {
  * Mock route for getting VC os images
  */
 router.get('/osimages', (req, res) => {
-    winston.info('Mock UI FETCH DATACENTERS (#' + mockOsImageRequestCounter + ')');
-    let osImageResponse = [];
-    if (mockOsImageRequestCounter > 0) {
-        osImageResponse = readFile('provider-vsphere-osimages.json');
+    const dc = '' + req?.query?.dc;
```
Comment (raymondz1) [inline cli/cmd/plugin/ui/web/tanzu-ui/node-server/src/routes/api/endpoints/vsphere.js:161]: do we need this empty string?

Comment (swalner-vmware) [reply cli/cmd/plugin/ui/web/tanzu-ui/node-server/src/routes/api/endpoints/vsphere.js:161]: I believe so, because otherwise `dc` can be undefined

Comment (swalner-vmware) [reply cli/cmd/plugin/ui/web/tanzu-ui/node-server/src/routes/api/endpoints/vsphere.js]: I removed it; I like it better without and whatever problem I was having before isn't there :)

**Review by miclettej — APPROVED** [review]
Looks great, thanks for the updates

**Pull request merged**